tests/scripts/icmp_responder.py

```diff
@@ -6,15 +6,21 @@
 from select import select
 
 
-def respond_to_icmp_request(socket, request):
+def respond_to_icmp_request(socket, request, dst_mac=None):
     """Respond to ICMP request."""
     reply = request.copy()
     reply[ICMP].type = 0
     # Force re-generation of the checksum
     reply[ICMP].chksum = None
     reply[IP].src, reply[IP].dst = request[IP].dst, request[IP].src
     reply[IP].chksum = None
-    reply[Ether].src, reply[Ether].dst = request[Ether].dst, request[Ether].src
+
+    if dst_mac is None:
+        icmp_reply_dst_mac = request[Ether].src
```

Comment:
I feel like one of these conditions will be obsolete, once heartbeats are sent with VLAN MAC instead of interface MAC.
When that happens, the original code should work as the `request[Ether].src` will have VLAN MAC in it.
Can a comment be added here for clarity?

Reply:
After further discussion with @tahmed-dev , this is going to be a permanent change to ICMP responder. In production scenarios, the heartbeat already is sent with the VLAN MAC so this PR is simply to bring the lab closer in line with production.